### ISSUE TITLE
composer fails to install / update

### ISSUE THREAD

**Issue by Sil3ntStorm** [PHPMailer/PHPMailer#1640]
# Problem description
Cannot install phpmailer using composer

# Code to reproduce
```
composer require phpmailer/phpmailer --no-update
composer update --no-dev
```

gives

```
[RuntimeException]
Could not read /var/www/vendor/phpmailer/phpmailer/composer.json
file_get_contents(/var/www/vendor/phpmailer/phpmailer/composer.json): failed to open stream: No such file or directory
```

using composer 1.8.0

**Comment by Synchro** [comment]
That suggests that your original installation did not work correctly. I’d recommend deleting your vendor folder and running `composer install` again.

**Comment by Sil3ntStorm** [comment]
Okay, so it works in an empty directory without any files.
However it does not work in a directory which already has a project in it.
```
composer create-project --no-dev --no-install silverstripe/installer .
composer require --no-update phpmailer/phpmailer
composer update --no-dev
```

Will bring up the error as described above.
Removing the `composer.lock` file as well as the `vendor/` directory then doing `composer install --no-dev` will exhibit the same behavior.

**Comment by Synchro** [comment]
OK, you're quite right here - `composer.json` was marked as `export-ignore` in `.gitattributes`, which stops it being exported when you check it out via composer, so you end up with that missing file problem. Since PHPMailer has no dependencies by default this doesn't stop it working, but it causes the issue you're seeing. It's a bit odd that this hasn't affected others, including me! I've pushed a fix to master in d065cd74b6ca1c5b0cacc0b67aa54c7b6fe992b4 - does that work for you (you'll probably need to change your dep to `dev-master` or `*` to pick it up until it's released)?

**Comment by Sil3ntStorm** [comment]
Yes, using dev-master works now. Thanks.

**Comment by Synchro** [comment]
This has been released as part of 6.0.7, so you don't need to use dev-master any more! Thanks for reporting this.
